pkg/controller/replicaset/replica_set_utils.go

```diff
@@ -115,10 +125,23 @@ func calculateStatus(rs *apps.ReplicaSet, filteredPods []*v1.Pod, manageReplicas
 			reason = "FailedDelete"
 		}
 		cond := NewReplicaSetCondition(apps.ReplicaSetReplicaFailure, v1.ConditionTrue, reason, manageReplicasErr.Error())
+		if !utilfeature.DefaultFeatureGate.Enabled(features.ExtendedWorkloadConditions) {
+			// old behaviour without LastUpdateTime field
+			cond.LastUpdateTime = metav1.NewTime(time.Time{})
+		}
 		SetCondition(&newStatus, cond)
 	} else if manageReplicasErr == nil && failureCond != nil {
```

Comment:
Removing the condition is a bit inconsistent to other conditions than just switch between `False`/`True`/`Unknown`. I suppose we probably can't / don't want to do something about it